extensions/azurePublishNew/src/node/azureResources/botChannel.ts

```diff
@@ -15,29 +36,49 @@ export const botRegistrationDefinition: ResourceDefinition = {
   group: AZURE_HOSTING_GROUP_NAME,
 };
 
-const getBotChannelProvisionMethod = (): ProvisionMethod => (
-  config: BotServiceConfig,
-  workingSet: ProvisionWorkingSet
-): Promise<ProvisionWorkingSet> => {
-  const appRegistrationResult: AppRegistrationResult = workingSet.appRegistration;
-  const webAppResult: WebAppResult = workingSet.webApp;
-  const { appId } = appRegistrationResult.appId;
-  const { endpoint } = webAppResult.endpoint;
-  const { hostname } = webAppResult.hostname;
+const botChannelProvisionMethod = (provisionConfig: ProvisionConfig): ProvisionMethod => {
+  const tokenCredentials = new TokenCredentials(provisionConfig.accessToken);
+  const azureBotService = new AzureBotService(tokenCredentials, provisionConfig.subscriptionId);
 
-  const provisionResult: BotChannelResult = { botName: '' };
+  return async (config: BotServiceConfig, workingSet: ProvisionWorkingSet): Promise<ProvisionWorkingSet> => {
+    const appId = workingSet.appRegistration?.appId;
+    const hostname = workingSet.webApp?.hostname;
+    const endpoint = `https://${hostname ?? config.hostname + '.azurewebsites.net'}/api/messages`;
 
-  return {
-    ...workingSet,
-    botChannelResult: provisionResult,
+    try {
+      await azureBotService.bots.create(config.resourceGroupName, config.displayName, {
+        properties: {
+          displayName: config.displayName,
+          endpoint: endpoint ?? '',
```

Comment:
There isn't any way for the endpoint to be undefined.  Consider checking above when constructing it.